src/screens/lesson/Lesson.js

```diff
@@ -0,0 +1,340 @@
+import React, { useState, useEffect } from "react";
+import { View, Text, TouchableOpacity, Image, ScrollView, Modal } from "react-native";
+import styles from "./styles";
+import HoverableView from "../../compoments/HoverableView";
+import { useNavigate } from "react-router-dom";
+import { AiFillHome, AiOutlineRead, AiOutlineVideoCamera, AiOutlineCheck } from "react-icons/ai";
+import { supabase } from "../../supabaseClient";
+import * as _ from 'lodash';
+
+export default function Lesson(props) {
+  const [subjectItems, setSubjectItems] = useState([]);
+  const [lessonItems, setLessonItems] = useState([]);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [lessonId, setLessonId] = useState(0);
+  const [lessonName, setLessonName] = useState(null);
+
+  let navigate = useNavigate();
+  const goToCourseScreen = () => {
+    navigate("/");
+  };
+
+  const onShowContent = (urlContent) => {
+    window.open(`${urlContent}`, '_blank');
+  };
+
+  const getSubjectItems = async () => {
+    const { data, error } = await supabase
+      .from('subjects')
```

Comment:
Why we need to get all subjects in DB? I expected we just get subjects of the current course.

Reply:
Edited
https://github.com/cam-drei/kids-study-front-end/pull/1/commits/d2cf0a4a70f0814d4b0f9014cccd48b8743ff02b